erpnext/accounts/doctype/pos_profile/pos_profile.js

```diff
@@ -42,18 +42,32 @@ frappe.ui.form.on('POS Profile', {
 			frm.toggle_display('offline_pos_section', is_offline);
 			frm.toggle_display('print_format_for_online', !is_offline);
 		});
+
+		frm.set_query('company_address', function(doc) {
+			if(!doc.company) {
+				frappe.throw(_('Please set Company'));
```

Comment:
Frappe uses double-underscores `__()` for translations in JS. The single variant is for server-side.